File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if (root==null) return new ArrayList<>();
        List<List<Integer>> ans = new LinkedList<>();
        
        LinkedList<TreeNode> que = new LinkedList<>();
        que.addLast(root);
        while(que.size()!=0) {
            int size = que.size();
            List<Integer> levelAns = new LinkedList<>();
            while(size-->0) {
                TreeNode rn = que.removeFirst();
                
                if(rn.left != null) {
                    que.addLast(rn.left);
                }
                
                if(rn.right != null) {
                    que.addLast(rn.right);
                }
                
                levelAns.add(rn.val);
            }
            ans.add(levelAns);
        }
        
        return ans;
    }
}

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(!root) return {};
         vector<vector<int>> ans;
        ///Declaration of que
        queue<TreeNode*> que;
        que.push(root);
        
        //WHILE LOOPS
        while(que.size()!=0) {
            int size = que.size();
            vector<int> levelAns;
            // Processing for each different level
            while(size-->0) {
                TreeNode* removeN = que.front(); que.pop();
                //setting up the next level in the que
                if(removeN->left) {
                    que.push(removeN->left);
                }
                
                if(removeN->right) {
                    que.push(removeN->right);
                }
                
                levelAns.push_back(removeN->val);
            }
            ans.push_back(levelAns);
        }
        return ans;
    }
};